pandas/core/indexing.py

```diff
@@ -1093,7 +1093,7 @@ def _getitem_iterable(self, key, axis=0):
         if is_bool_indexer(key):
             key = check_bool_indexer(labels, key)
             inds, = key.nonzero()
-            return self.obj.take(inds, axis=axis, convert=False)
+            return self.obj.take(inds, axis=axis)
```

Comment:
so this use of public ``.take()`` is fine (we are in a a not-core class), though this is arguable as we use internal methods in Index all the time.

Reply:
We'll see.  The reason why the conversion process is taking some time is because there some usages that cause things to break in the tests.

Reply:
and all the more reason to change it